src/components/FlexibleForm/ControlDecorators/withRepeatableActions/withRepeatableActions.js

```diff
@@ -39,34 +39,44 @@ export const withRepeatableActions = memo(props => {
   }));
 
   const needsTranslation = wrapperLabel && !isFormattedMessage(wrapperLabel) && isTranslationId(wrapperLabel);
-  const hasRecords = !isEmpty(referenceTable);
 
   return (
-    <div className={classNames(styles.decorator, isEmpty(legend) ? styles['no-legend'] : '')}>
-      {enabled && hasRecords && (
-        <div data-test-repeatable-decorator>
-          {needsTranslation ? (
-            <FormattedMessage id={wrapperLabel}>
-              {placeholder => (
-                <Field
-                  name={wrapperFieldName}
-                  component={Select}
-                  itemToString={identity}
-                  dataOptions={dataOptions}
-                  placeholder={placeholder}
-                />
-              )}
-            </FormattedMessage>
-          ) : (
-            <Field
-              name={wrapperFieldName}
-              component={Select}
-              itemToString={identity}
-              dataOptions={dataOptions}
-              placeholder={wrapperLabel}
-            />
+    <div className={styles.decorator}>
+      {enabled && (
+        <>
+          {legend && (
+          <Headline
+            tag="h3"
+            margin="xx-small"
+          >
+            <FormattedMessage id={legend} />
+          </Headline>
           )}
```

Comment:
Consider moving Headline to a variable
```suggestion
          {withLegend && legend}
```